Minor changes for compilation

tathanhdinh opened the pull request:
Travis builds now.

iankronquist commented on scripts/ci.sh:
```diff
@@ -5,5 +5,5 @@ set -e
-cargo clippy -- -D clippy || true
-cargo fmt -- --write-mode diff
+cargo clippy -- -D clippy::all || true
+cargo fmt -- --check || true
```
Can you remove second "|| true" but leave the first one? We want the build to fail if cargo fmt fails, but we want it to pass if clippy is absent because the nightly build of clippy broke for reasons beyond our control (which has happened occasionally in the past).

tathanhdinh replied:
Sorry, I've used a different format config on my machine and too lazy to check on Travis.

iankronquist commented on src/runtime.rs:
```diff
@@ -46,13 +47,15 @@ pub extern "C" fn _Unwind_Resume() {
-pub extern "C" fn panic_fmt(fmt: fmt::Arguments, file: &'static str, line: u32) -> ! {
-
-    error!("PANIC: {} {} {}\n", fmt, file, line);
+#[panic_handler]
```
Thanks for fixing this. I've been meaning to get around to it.

iankronquist commented on src/vmx.rs:
```diff
@@ -366,68 +377,75 @@ pub fn rdmsrl(msr: MSR) -> u64 {
+        );
     }
-    ((edx as u64) << 32) | (eax as u64)
+    (u64::from(edx) << 32) | u64::from(eax)
```
I had to look this up, and this looks like a good change.

iankronquist replied:
Thanks

iankronquist commented on src/vmx.rs:
```diff
@@ -798,17 +825,15 @@ fn prepare_vmx_memory_region(vmx_region: *mut u8, vmx_region_size: usize) {
     for i in mem::size_of::<u32>()..vmx_region_size {
         unsafe {
-            *vmx_region.offset(i as isize) = 0;
+            vmx_region.add(i).write(0);
```
This change is good, but there's probably a better way to write "memset(&vmx_region[sizeof(u32)], 0, PAGE_SIZE-sizeof(u32))" in rust these days. This is a note to self mostly, no need to change anything.

tathanhdinh replied:
Thank you for notifying this, I found that there is `core::ptr::write_bytes` which handles this better.

iankronquist commented on src/vmx.rs:
```diff
@@ -894,13 +913,9 @@ pub fn load_vm(vmcs: *mut u8, vmcs_phys: u64, vmcs_size: usize) -> Result<(), ()
-            }
+    if let Ok(vmcs_phys) = vmptrst() {
+        if let Err(code) = vmclear(vmcs_phys) {
+            error!("vmclear failed with error code {}", code);
```
Much cleaner, thanks.

tathanhdinh commented:
@iankronquist Thank you for taking time to review the PR. I'm very happy to know that it captures your attention.

Sorry for having forced the result of `rustfmt` to be always `true` (that is because I've used another configuration while making the PR, then `rustfmt` of Travis complains, too lazy to check the problem so I've added `|| true`). I've changed it.

A minor change in the updated PR is to use `cargo xbuild` instead of `xargo` (I think they are quite the same in this case, but correct me if I was wrong).

The change `vmx_region.add(i).write(0);` is indeed not nice, since there is `mem::ptr::write_bytes` which handles this better.

Again, thank you for encouraging feedbacks.

iankronquist commented:
Thanks for your help! I know this is like a new year, and I'm a terrible maintainer, but I think it's really cool to have people come by and help on my little hobby project. If you want to make more changes to the project I'll try to be more responsive in the future.
